homeassistant/components/yamaha_musiccast/__init__.py

```diff
@@ -190,3 +199,36 @@ def device_info(self) -> DeviceInfo:
             device_info["via_device"] = (DOMAIN, self.coordinator.data.device_id)
 
         return device_info
+
+
+class MusicCastCapabilityEntity(MusicCastDeviceEntity):
+    """Base Entity type for all capabilities."""
+
+    def __init__(
+        self,
+        coordinator: MusicCastDataUpdateCoordinator,
+        capability: Capability,
+        zone_id: str = None,
+    ) -> None:
+        """Initialize the switch."""
+        if zone_id is not None:
+            self._zone_id = zone_id
+        self.capability = capability
+        super().__init__(name=capability.name, icon="", coordinator=coordinator)
+        self._attr_entity_category = ENTITY_CATEGORY_MAPPING.get(capability.entity_type)
+
+    async def async_added_to_hass(self):
+        """Run when this Entity has been added to HA."""
+        await super().async_added_to_hass()
+        # Sensors should also register callbacks to HA when their state changes
```

Comment:
Are these sensors?

Reply:
We used the word sensor as a synonym for entity in some comments/doc strings all over the component. I believe this was not a big deal until now, but as we want to add other entity types such as select, switch and sensor with future PRs this could now lead to confusion and I will refactor that in another PR for already existing code.